.ruff.toml

```diff
@@ -0,0 +1,26 @@
+# [tool.ruff]  # <- TODO Uncomment when migrating to pyproject.toml
+exclude = [
+    ".eggs",
+    ".tox",
+    "build",
+    "ci/templates",
+    "dist",
+]
+line-length = 140
+select = [
```

Comment:
Shouldn't this be `extend-select` instead?

Reply:
> If you're wondering how to configure Ruff, here are some recommended guidelines:
> * Prefer select and ignore over extend-select and extend-ignore, to make your rule set explicit.

https://beta.ruff.rs/docs/configuration

Reply:
Well all I want is to not hardcode a ruleset (which `select = [stuff]` exactly is). I don't think any of those recommendations are worth considering blindly given the tool's age and popularity.

How about `select = ["ALL"]` and ignore whatever doesn't make sense for pytest-cov?

Reply:
> If you're wondering how to configure Ruff, here are some __recommended guidelines__:
> *    Prefer select and ignore over extend-select and extend-ignore, to make your rule set explicit.
> *    Use `ALL` with discretion. Enabling `ALL` will implicitly enable new rules whenever you upgrade.
> *    Start with a small set of rules (`select = ["E", "F"]`) and add a category at-a-time. For example, you might consider expanding to `select = ["E", "F", "B"]` to enable the popular flake8-bugbear extension.
> *    By default, Ruff's __autofix is aggressive__. If you find that it's too aggressive for your liking, consider turning off autofix for specific rules or categories (see [FAQ](https://beta.ruff.rs/docs/faq/#ruff-tried-to-fix-something--but-it-broke-my-code)).

Reply:
Ok so I've played with it a bit more.... now I am not really sure what the best way... eg: these are either invalid checks or would be a pain to sort out, or not even worth fixing:
```
ignore = [
    "D", # pydocstyle
    "ANN", # flake8-annotations
    "ARG", # flake8-unused-arguments
    "BLE", # flake8-blind-except
    "C90", # mccabe
    "EM", # flake8-errmsg
    "FBT", # flake8-boolean-trap
    "INP", # flake8-no-pep420
    "PLR", # Pylint Refactor
    "PLW", # Pylint Warning
    "Q", # flake8-quotes
    "RET", # flake8-return
    "SIM102", # flake8-simplify collapsible-if
    "SIM105", # flake8-simplify use-contextlib-suppress
    "SLF", # flake8-self
    "T20", # flake8-print
    "S101", # flake8-bandit assert
    "S102", # flake8-bandit exec
    "S110", # flake8-bandit try-except-pass
    "TRY", # tryceratops
]
```

Now considering that rsync features are being dropped in xdist ... a lot of bulk using that in pytest-cov becomes unnecessary, if it isn't already as some people are suggesting pytest-cov is not necessary for getting xdist support...

I guess select with a few is fine for now.

Reply:
So...  How should we proceed?  Given your explorations and your knowledge of the codebase, do you want to open a new PR that cherry-picks from this one?  I have no trouble closing this with unmerged commits.  Or should I press on?

Reply:
I'd also recommend the `C4` checks as they are popular, increase performance, and have autofixes enabled for all rules.